src/main/scala/cognite/spark/v1/DefaultSource.scala

```diff
@@ -165,6 +165,26 @@ class DefaultSource
       val relation = new AssetHierarchyBuilder(config)(sqlContext)
       relation.buildFromDf(data)
       relation
+    } else if (resourceType == "datapoints" || resourceType == "stringdatapoints") {
+      val relation = resourceType match {
+        case "datapoints" =>
+          new NumericDataPointsRelationV1(config)(sqlContext)
+        case "stringdatapoints" =>
+          new StringDataPointsRelationV1(config)(sqlContext)
+      }
+      if (config.onConflict == OnConflict.Delete) {
+        // Datapoints support 100_000 per request when inserting, but only 10_000 when deleting
+        val batchSize = config.batchSize.getOrElse(Constants.DefaultDataPointsLimit)
+        data.foreachPartition((rows: Iterator[Row]) => {
+          import CdpConnector._
```

Comment:
Some implicit for the partTraverse_ , right? Perhaps we can import the specific one.

Reply:
good idea